File: typescript/6kyu/Decode_the_Morse_code.ts
// import { MORSE_CODE } from "./preloaded";
const decodeMorse = (morseCode: string): string => {
  return morseCode
    .split(" ")
    .map((v, i) => (MORSE_CODE[v] != undefined ? MORSE_CODE[v] : 1))
    .join("")
    .replace(/11/g, " ")
    .replace(/1/g, " ")
    .trim();
};

console.log(decodeMorse(".... . -.--   .--- ..- -.. ."));

File: rust/src/6kyu/Decode_the_Morse_code.rs
mod preloaded;
use preloaded::MORSE_CODE;
// MORSE_CODE is `HashMap<String, String>`. e.g. ".-" -> "A".

fn decode_morse(encoded: &str) -> String {
    encoded
    .split(' ')
    .map(|x| { MORSE_CODE.get(x).unwrap_or(&" ".to_string()).clone() })
    .collect::<String>()
    .split_whitespace()
    .collect::<Vec<_>>()
    .join(" ")
}